R/create_demographic_lookup.R

```diff
@@ -0,0 +1,276 @@
+            )) > 0 |
+            .data$spec == "AB" |
+            .data$sigfac %in% c("1E", "1D") |
+            .data$recid == "GLS"),
+
+      # Maternity classification
+      maternity_cohort = .data$recid == "02B",
+
+      # High CC classification
+      # FOR FUTURE: PhysicalandSensoryDisabilityClientGroup or LearningDisabilityClientGroup = "Y",
+      # then high_cc_cohort = TRUE
+      # FOR FUTURE: Care home removed, here's the code: .data$recid = "CH" & age < 65
+      high_cc_cohort =
+        rowSums(
+          dplyr::across(
+            c(.data$dementia, .data$hefailure, .data$refailure, .data$liver, .data$cancer)
+          )
+        ) >= 1 |
+          .data$spec == "G5",
+
+      # Medium CC classification
+      medium_cc_cohort = rowSums(
+        dplyr::across(
+          c(.data$cvd, .data$copd, .data$chd, .data$parkinsons, .data$ms)
+        )
+      ) >= 1,
+
+      # Low CC classification
+      low_cc_cohort = rowSums(
+        dplyr::across(
+          c(.data$epilepsy, .data$asthma, .data$arth, .data$diabetes, .data$atrialfib)
+        )
+      ) >= 1,
+
+      # Note from SPSS: we could add CMH here
+
+      # Assisted living in the Community
+      # Not using this cohort until we have more datasets and Scotland complete DN etc.
+      # Code: .data$recid %in% c('HC-', 'HC + ', "RSP", "DN", "MLS", "INS", "CPL", "DC")
+      comm_living_cohort = FALSE,
+
+      # Seperate out prescribing cost for major conditions
+      prescribing_cost = dplyr::if_else(.data$recid == "PIS", .data$cost_total_net, 0),
+
+      # Adult Major Conditions classification
+      adult_major_cohort = .data$age >= 18 & (.data$prescribing_cost >= 500 | .data$recid == "01B"),
+
+      # Child Major Conditions classification
+      child_major_cohort = .data$age < 18 & (.data$prescribing_cost >= 500 | .data$recid == "01B"),
+
+      # Make sure any cohorts not assigned return FALSE
+      dplyr::across(.data$mh_cohort:.data$child_major_cohort, ~ tidyr::replace_na(., FALSE))
+    )
+
+  return(return_data)
+}
+
+#' Assign End of Life cohort based on death codes
+#'
+#' @param data A data frame containing recid and the eleven death diagnosis variables
+#'
+#' @return A data frame with variables for external causes of death and members of the EoL cohort
+#'
+#' @family Demographic and Service Use Cohort functions
+assign_eol_cohort <- function(data) {
+  check_variables_exist(data, variables = c(
+    "recid", "deathdiag1", "deathdiag2", "deathdiag3", "deathdiag4", "deathdiag5",
+    "deathdiag6", "deathdiag7", "deathdiag8", "deathdiag9", "deathdiag10",
+    "deathdiag11"
+  ))
+
+  external_deaths <- c(
+    glue::glue("V{stringr::str_pad(1:99, 2, 'left', '0')}"),
+    glue::glue("W{stringr::str_pad(0:99, 2, 'left', '0')}"),
+    glue::glue("X{stringr::str_pad(0:99, 2, 'left', '0')}"),
+    glue::glue("Y{stringr::str_pad(0:84, 2, 'left', '0')}")
+  )
+  falls_codes <- c(glue::glue("W{stringr::str_pad(0:19, 2, 'left', '0')}"))
```

Comment:
Add comments for these to show what the code should be generating e.g. V01-V99 etc.

Reply:
Done